### PR TITLE
feat: adapt Kafka bindings to v3

### PR DESCRIPTION
**Pull request by lbroudoux**
**Description**

This is the companion PR to https://github.com/asyncapi/bindings/pull/221.

As discussed and proposed [here](https://github.com/asyncapi/bindings/issues/182#issuecomment-1792837812) in issue #182, this PR: 
- Duplicates previous version `0.4.0` in `v2-0.4.0` and `v3-0.4.0` to keep track of bindings changes related to AsyncAPI v3 or v3
- Schemas related to v2 are referencing the v2 specification extensions schema (https://raw.githubusercontent.com/asyncapi/spec-json-schemas/v2.14.0/schemas/2.4.0.json#/definitions/specificationExtension)
- Schemas related to v3 are referencing the v3 specification extension schema (http://asyncapi.com/definitions/3.0.0/specificationExtension.json)

Another PR with specific schemas for v2 and v3 is on its way on `asyncapi/spec-json-schemas`.

**Related issue(s)**

Relates to https://github.com/asyncapi/bindings/issues/182

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@lbroudoux you need to re-target the PR to `next-major-spec` 🙂

**Comment by lbroudoux** [comment]
Oops...

**Review by jonaslagoni — COMMENTED** [review]
In order for the bindings to be validated as well, you need to adapt: https://github.com/asyncapi/spec-json-schemas/blob/9639ca9bf25b5bb207fd8a1b37f1c37a3c39ed4e/definitions/3.0.0/channelBindingsObject.json#L84

As well for message, operation and server 🙂

**Comment by lbroudoux** [comment]
Tests are failing but seem normal as http://asyncapi.com/bindings/kafka/v3-0.4.0/server.json is not yet deployed and available.

File: definitions/3.0.0/channelBindingsObject.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
     "kafka": {
       "properties": {
         "bindingVersion": {
-          "enum": ["0.4.0", "0.3.0"]
+          "enum": ["v3-0.4.0", "0.4.0", "0.3.0"]
```
Comment (jonaslagoni) [inline definitions/3.0.0/channelBindingsObject.json:87]: You still want the other binding versions to be used for v3? 🤔

**Review by dalelane — APPROVED** [review]
thanks for doing this!

File: definitions/3.0.0/channelBindingsObject.json (hunk 1 above)
Comment (lbroudoux) [reply definitions/3.0.0/channelBindingsObject.json:87]: Considering `0.4.0` (or `v3-0.4.0`) is the latest, I think it's ok to still allow the previous one and allow decoupled spec + binding upgrade.

**Comment by lbroudoux** [comment]
Now replaced by https://github.com/asyncapi/spec-json-schemas/pull/453

Could you all confirm we can cancel this one?

**Comment by dalelane** [comment]
> Now replaced by #453
> 
> Could you all confirm we can cancel this one?

yeh, makes sense to me

**Pull request closed without merging**